src/runtime/Types/ExceptionClassObject.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+
+namespace Python.Runtime;
+
+/// <summary>
+/// Base class for Python types that reflect managed exceptions based on
+/// System.Exception
+/// </summary>
+/// <remarks>
+/// The Python wrapper for managed exceptions LIES about its inheritance
+/// tree. Although the real System.Exception is a subclass of
+/// System.Object the Python type for System.Exception does NOT claim that
+/// it subclasses System.Object. Instead TypeManager.CreateType() uses
+/// Python's exception.Exception class as base class for System.Exception.
```

Comment:
NIT: this is not specific to `System.Exception` anymore, any type can mix-in any compatible Python base class. The one for exception is handled in the default base type provider. E.g. this remark can be removed.